lib/src/account-creation/account-creation.widget.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:flutter/cupertino.dart';
+import 'package:nice_flutter_kit/nice_flutter_kit.dart';
+import 'package:nice_flutter_kit/src/account-creation/widgets/nice-account-creation-buttons.widget.dart';
+import 'package:provider/provider.dart';
+
+class NiceAccountCreation extends StatefulWidget {
+  /// Config used for the account creation
+  final NiceAccountCreationConfig config;
+
+  /// Controller used for the page view
+  /// This can be used to change pages from outside of the [NiceAccountCreation] widget, for example go to the previous
+  /// page when the back button of an app bar is pressed
+  /// If not null, this widget won't dispose it
+  final PageController? pageController;
+
+  /// Pages that will be displayed in the account creation page view
+  final List<NiceAccountCreationBasePage> pages;
+
+  /// Called whenever the current [PageView] page changes
+  /// [index] represents the index of the page in the [PageView]
+  /// [realIndex] represent index of the page in [pages]
+  /// These indexes can differ if we are hiding pages
+  final void Function(int index, int realIndex)? onPageChange;
+
+  /// Called when the user is at the last page and presses the "next" button
+  final VoidCallback onSubmit;
+
+  NiceAccountCreation({
+    required this.config,
+    this.pageController,
+    required this.pages,
+    this.onPageChange,
+    required this.onSubmit,
+  }) : assert(pages.isNotEmpty);
+
+  @override
+  State<NiceAccountCreation> createState() => _NiceAccountCreationState();
+}
+
+class _NiceAccountCreationState extends State<NiceAccountCreation> {
+  late final _localPageController;
+
+  PageController get _pageController => widget.pageController ?? _localPageController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (widget.pageController == null) {
+      _localPageController = PageController();
+    }
+  }
+
+  @override
+  void dispose() {
+    if (widget.pageController == null) {
+      // Only dispose of the PageController if it was created by this widget
+      _localPageController.dispose();
+    }
```

Comment:
Ça me semble plus safe de faire ça?
```suggestion
    if (_localPageController != null) {
      // Only dispose of the PageController if it was created by this widget
      _localPageController.dispose();
    }
```